pkg/watcherapi/statefulset.go

```diff
@@ -130,6 +132,12 @@ func StatefulSet(
 		}
 	}
 
+	// add MTLS cert if defined
```

Comment:
You need to modify https://github.com/openstack-k8s-operators/watcher-operator/blob/cef7a90ee8f2fe8ee1d3ca4597c17af679e90afd/templates/watcherapi/config/watcher-api-config.json to copy the certificates to the expected location. Somethin like:

https://github.com/openstack-k8s-operators/nova-operator/blob/e890fbdf87267967a9b1c47f4446e3fd0edfe593/templates/novametadata/config/nova-metadata-config.json#L51-L66

Same applies to the watcherapplier and watcherdecisionengine.